### PR TITLE
Added psp

### PR DESCRIPTION
**Pull request by mattvella07**

### CODE REVIEW

File: charts/vault-operator/templates/role.yaml [hunk 1]
```diff
@@ -77,3 +77,11 @@ rules:
...
   - create
   - watch
+- apiGroups:
+    - extensions
+  resources:
+    - podsecuritypolicies
+  verbs:
+    - use
+  resourceNames:
+    - default
```
Comment (naemono) [inline charts/vault-operator/templates/role.yaml:87]: If we're going to submit this upstream, all of this needs to be configurable.  The security context pieces should be configurable from the values file, and the psp to use also needs to be in the values, and be a bool to turn on/off

Comment (mattvella07) [reply charts/vault-operator/templates/role.yaml:87]: Good idea, I'll update it

File: charts/vault-operator/templates/role.yaml [hunk 2]
```diff
@@ -77,3 +77,13 @@ rules:
...
   - watch
+{{- if .Values.rbac.psp.enabled }}
+- apiGroups:
+    - extensions
+  resources:
+    - podsecuritypolicies
+  verbs:
+    - use
+  resourceNames:
+    - default
```
Comment (naemono) [inline charts/vault-operator/templates/role.yaml:88]: The name of this sec policy needs to be configurable also

Comment (mattvella07) [reply charts/vault-operator/templates/role.yaml:92]: 👍

File: charts/vault-operator/templates/role.yaml (hunk 1 above)
Comment (naemono) [reply charts/vault-operator/templates/role.yaml:92]: You've got to remove this one

File: charts/vault-operator/values.yaml [hunk 3]
```diff
@@ -70,3 +70,13 @@ etcd-operator:
     commandArgs:
       cluster-wide: true
       create-crd: false
+
+rbac:
+  psp:
+    enabled: true
+    name:
```
Comment (naemono) [inline charts/vault-operator/values.yaml:77]: Provide a default here

Comment (mattvella07) [reply charts/vault-operator/values.yaml]: Good idea

File: charts/vault-operator/templates/role.yaml (hunk 1 above)
Comment (naemono) [reply charts/vault-operator/templates/role.yaml:92]: This and the line above need removed

**Pull request merged**